crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -403,34 +438,23 @@ export.finalize_transaction.4
     hmerge
     # => [ACCOUNT_UPDATE_COMMITMENT]
 
-    # prepare stack for build_output_vault by loading fee asset from local
-    padw loc_loadw.0
-    # => [FEE_ASSET, ACCOUNT_UPDATE_COMMITMENT]
-
-    # build the output vault
-    exec.build_output_vault
-    # => [ACCOUNT_UPDATE_COMMITMENT]
-
-    # compute output notes commitment
-    exec.note::compute_output_notes_commitment
-    # => [OUTPUT_NOTES_COMMITMENT, ACCOUNT_UPDATE_COMMITMENT]
-
-    # copy output note data to the advice map
-    exec.copy_output_notes_to_advice_map
-    # => [OUTPUT_NOTES_COMMITMENT, ACCOUNT_UPDATE_COMMITMENT]
+    # ------ Assert assets are preserved ------
 
     # assert no net creation or destruction of assets over the transaction
     exec.memory::get_input_vault_root exec.memory::get_output_vault_root
     assert_eqw.err=ERR_EPILOGUE_TOTAL_NUMBER_OF_ASSETS_MUST_STAY_THE_SAME
-    # => [OUTPUT_NOTES_COMMITMENT, ACCOUNT_UPDATE_COMMITMENT]
+    # => [ACCOUNT_UPDATE_COMMITMENT]
```

Comment:
We should be able to do this right after the output vault has been updated on line 359 above, right? Would it make sense to move this up to make it closer to that computation? May make the code a bit more linear and easier to follow.